internals/daemon/api_files_test.go

```diff
@@ -446,15 +446,17 @@ func (s *filesSuite) TestMakeDirsMultiple(c *C) {
 
 func (s *filesSuite) TestMakeDirsUserGroupMocked(c *C) {
 	type args struct {
-		path string
-		perm os.FileMode
-		uid  sys.UserID
-		gid  sys.GroupID
-	}
-	var mkdirChownCalls []args
-	mkdirChown = func(path string, perm os.FileMode, uid sys.UserID, gid sys.GroupID) error {
-		mkdirChownCalls = append(mkdirChownCalls, args{path, perm, uid, gid})
-		return os.Mkdir(path, perm)
+		path    string
+		perm    os.FileMode
+		options osutil.MkdirOptions
+	}
+	var mkdirCalls []args
+	mkdir = func(path string, perm os.FileMode, options *osutil.MkdirOptions) error {
+		if options == nil {
+			options = &osutil.MkdirOptions{}
+		}
+		mkdirCalls = append(mkdirCalls, args{path, perm, *options})
+		return os.MkdirAll(path, perm)
```

Comment:
I realize it is what was done before, but why do we actually create the directory when we're mocking this? Can't we just test the mock calls, or does the test require that the directory actually be created to work?

Reply:
Without making the directory, it would fail. This is because the test checks the result [here](https://github.com/canonical/pebble/blob/master/internals/daemon/api_files_test.go#L526).